controllers/authController.js

```diff
@@ -0,0 +1,58 @@
+const User = require('../models/User');
+const CustomError = require('../errors');
+//const { attachCookiesToResponse, createTokenUser } = require('../utils');
+
+const register = async (req, res) => {
+	const { email, name, password } = req.body;
+
+	const emailAlreadyExists = await User.findOne({ email });
+	if (emailAlreadyExists) {
+		throw new CustomError.BadRequestError('Email already exists');
+	}
+
+	const isFirstAccount = (await User.countDocuments({})) === 0;
+	const role = isFirstAccount ? 'admin' : 'user';
+
+	const user = await User.create({ name, email, password, role });
+	const tokenUser = createTokenUser(user);
+	attachCookiesToResponse({ res, user: tokenUser });
+	res.status(200).json({ user: tokenUser });
+};
+const login = async (req, res) => {
+	const { email, password } = req.body;
+
+	if (!email || !password) {
+		throw new CustomError.BadRequestError('Please provide email and password');
+	}
+	const user = await User.findOne({ email });
+
+	if (!user) {
+		throw new CustomError.BadRequestError('Invalid Credentials');
+	}
+	const isPasswordCorrect = await user.comparePassword(password);
+	if (!isPasswordCorrect) {
+		throw new CustomError.BadRequestError('Invalid Credentials');
+	}
+	//const tokenUser = createTokenUser(user);
+	//attachCookiesToResponse({ res, user: tokenUser });
+
+	res.status(200).json({ user: 'blah-blah' });
+};
+const logout = async (req, res) => {
+	res.cookie('token', 'logout', {
+		httpOnly: true,
+		expires: new Date(Date.now() + 1000),
+	});
+	res.status(200).json({ msg: 'user logged out!' });
+};
+
+const test = async (req, res) => {
+	res.status(200).json({ msg: 'Route is Up!' });
```

Comment:
This is just a set-up of the repo, but these return objects ain't gonna cut it. You decide whether to leave them for now. I'd change them to something more appropriate (see: docs).

Reply:
Fixed :)